packages/http-client-csharp/eng/scripts/Build-Packages.ps1

```diff
@@ -90,7 +90,7 @@ $mgcCustomizationVersion = Get-CsprojVersion -csprojFilePath "$packageRoot/gener
 
 if ($BuildNumber) {
     # set package versions
-    $versionTag = $Prerelease ? "-alpha" : "-beta"
+    $versionTag = $Prerelease ? ($PublishType -eq 'public' ? "-alpha" : '-dev') : "-beta"
```

Comment:
We don't use dev for nuget packages. I understand the concern about publishing alphas of the emitter from PRs, but I don't think we have the same concern for the nuget package as this is already in a private dev feed.

Reply:
The concern is, the NuGet package published to private dev feed from main branch is using alpha.
How can we distinguish from the NuGet packages published from PR?

Reply:
There isn't really a need to distinguish between the two. They are both alphas and should be treated as such. They are also limited to the private dev feed. We can have breaking changes to main and in PRs in these versions. The emitter is different since those are published to a public feed. This is also the main entry point for consumers. The only customers using the Nuget dev feed would be plugin authors and they would need to understand that there can be breaking changes between alpha versions, regardless of whether it is from a PR or a merge to main.

Reply:
Could you check https://github.com/microsoft/typespec/issues/4507#issuecomment-2406439499? ideally, we should release to public from main as well, like npm package for emitter

Reply:
I do think we need to distinguish between the two.
If something went wrong in the PR alpha, it would cause failure for other people.

Reply:
Let's discuss this at our next sync on Tuesday.

Reply:
> we should release to public NuGet from main as well, like npm package for emitter

We would release betas and GA versions of the packages to public Nuget, but the alpha versions would only be published to the dev feed, similar to what we do for all of the Azure SDK packages.

Reply:
> We would release betas and GA versions of the packages to public Nuget, but the alpha versions would only be published to the dev feed, similar to what we do for all of the Azure SDK packages.

Agreed. `-dev` is kind of a workaround for now to distinguish release from PR and main branch since both of them are `-alpha`.

Reply:
reverted